spec.bs

```diff
@@ -1368,26 +1373,36 @@ To <dfn>build bid generators map</dfn> given an [=auction config=] |auctionConfi
       * |ig|'s [=interest group/bidding url=] is null;
       * |ig|'s [=interest group/ads=] is null, or [=list/is empty=].
     1. Let |signalsUrl| be |ig|'s [=interest group/trusted bidding signals url=].
+    1. Let |slotSizeQueryParam| be the result of [=calculating the ad slot size query param=]
+      given |ig| and |auctionConfig|.
     1. Let |joiningOrigin| be |ig|'s [=interest group/joining origin=].
     1. If |bidGenerators| does not [=map/contain=] |buyer|:
       1. Let |perBuyerGenerator| be a new [=per buyer bid generator=].
       1. Let |perSignalsUrlGenerator| be a new [=per signals url bid generator=].
       1. [=map/Set=] |perSignalsUrlGenerator|[|joiningOrigin|] to « |ig| ».
-      1. [=map/Set=] |perBuyerGenerator|[|signalsUrl|] to |perSignalsUrlGenerator|.
+      1. [=map/Set=] |perSlotSizeQueryParam|[|slotSizeQueryParam|] to |perSignalsUrlGenerator|.
+      1. [=map/Set=] |perBuyerGenerator|[|signalsUrl|] to |perSlotSizeQueryParam|.
       1. [=map/Set=] |bidGenerators|[|buyer|] to |perBuyerGenerator|.
       1. TODO: add a perBiddingScriptUrlGenerator layer that replaces the list of IGs with a map
         from biddingScriptUrl to a list of IGs.
     1. Otherwise:
       1. Let |perBuyerGenerator| be |bidGenerators|[|buyer|].
-      1. If |perBuyerGenerator| does not [=map/contain=] |signalsUrl|:
+      1. If |perSlotSizeQueryParam| does not [=map/contain=] |signalsUrl|:
         1. Let |perSignalsUrlGenerator| be a new [=per signals url bid generator=].
         1. [=map/Set=] |perSignalsUrlGenerator|[|joiningOrigin|] to « |ig| ».
-        1. [=map/Set=] |perBuyerGenerator|[|signalsUrl|] to |perSignalsUrlGenerator|.
+        1. [=map/Set=] |perSlotSizeQueryParam|[|slotSizeQueryParam|] to |perSignalsUrlGenerator|.
+        1. [=map/Set=] |perBuyerGenerator|[|signalsUrl|] to |perSlotSizeQueryParam|.
       1. Otherwise:
-        1. Let |perSignalsUrlGenerator| be |perBuyerGenerator|[|signalsUrl|].
-        1. If |perSignalsUrlGenerator| does not [=map/contain=] |joiningOrigin|, then [=map/set=]
-          |perSignalsUrlGenerator|[|joiningOrigin|] to « |ig| ».
-        1. Otherwise, [=list/append=] |ig| to |perSignalsUrlGenerator|[|joiningOrigin|].
+        1. Let |perSlotSizeQueryParam| be |perBuyerGenerator|[|signalsUrl|].
+        1. If |perSlotSizeQueryParam| does not [=map/contain=] |slotSizeQueryParam|:
+          1. Let |perSignalsUrlGenerator| be a new [=per signals url bid generator=].
+          1. [=map/Set=] |perSignalsUrlGenerator|[|joiningOrigin|] to « |ig| ».
+          1. [=map/Set=] |perSlotSizeQueryParam|[|slotSizeQueryParam|] to |perSignalsUrlGenerator|.
+        1. Otherwise:
+          1. Let |perSignalsUrlGenerator| be |perSlotSizeQueryParam|[|perSlotSizeQueryParam|].

```

Comment:
[|perSlotSizeQueryParam|] -> [|slotSizeQueryParam|].

Not related to this PR. (This [=build bid generators map=] method here is very complicated with a deeply nested structure, and pretty hard to read. I need to think about if there's a better way to deal with these generators).

Reply:
Done.  Yea, this is really difficult to follow (and was hard to modify).  Would love to see it cleaned up a bit.